extension/js/common/api/email-provider/gmail/google.ts

```diff
@@ -41,22 +41,31 @@ export class Google {
     return await GoogleAuth.apiGoogleCallRetryAuthErrorOneTime(acctEmail, request) as RT;
   };
 
-  public static contactsGet = async (acctEmail: string, query?: string, progress?: ProgressCbs, max: number = 10, start: number = 0) => {
+  public static contactsGet = async (acctEmail: string, query?: string, progress?: ProgressCbs, max: number = 10) => {
     progress = progress || {};
     const method = 'GET';
     const contentType = 'application/json; charset=UTF-8';
-    const url = `${GOOGLE_API_HOST}/m8/feeds/contacts/default/thin`;
-    const data = { 'alt': "json", 'q': query, 'v': '3.0', 'max-results': max, 'start-index': start };
+    const searchContactsUrl = `${PEOPLE_API_HOST}/v1/people:searchContacts`;
+    const searchOtherContactsUrl = `${PEOPLE_API_HOST}/v1/otherContacts:search`;
+    const data = { query, 'readMask': 'names,emailAddresses', 'pageSize': max };
     const xhr = Api.getAjaxProgressXhrFactory(progress);
     const headers = { 'Authorization': await GoogleAuth.googleApiAuthHeader(acctEmail) };
-    const contacts = await GoogleAuth.apiGoogleCallRetryAuthErrorOneTime(acctEmail,
-      { xhr, url, method, data, headers, contentType, crossDomain: true, async: true }) as GmailRes.GoogleContacts;
-    return contacts.feed.entry && contacts.feed.entry // todo - causes weird function signature, could be improved to return empty arr
-      .filter(entry => !!(entry.gd$email || []).find(email => email.primary === "true")) // find all entries that have primary email
-      .map(e => ({
-        email: (e.gd$email || []).find(e => e.primary === "true")!.address,
-        name: e.gd$name && e.gd$name.gd$fullName && e.gd$name.gd$fullName.$t
-      }));
+    const contacts = await Promise.all([
+      GoogleAuth.apiGoogleCallRetryAuthErrorOneTime(acctEmail,
+        { xhr, url: searchContactsUrl, method, data, headers, contentType, crossDomain: true, async: true }) as Promise<GmailRes.GoogleContacts>,
+      GoogleAuth.apiGoogleCallRetryAuthErrorOneTime(acctEmail,
+        { xhr, url: searchOtherContactsUrl, method, data, headers, contentType, crossDomain: true, async: true }) as Promise<GmailRes.GoogleContacts>
+    ]);
+    const primaryContacts = contacts[0].results || [];
+    const otherContacts = contacts[1].results || [];
+    const contactsMerged = [...primaryContacts, ...otherContacts];
+    return contactsMerged
+      .filter(entry => !!(entry.person?.emailAddresses || []).find(email => email.metadata.primary === true)) // find all entries that have primary email
+      .map(entry => {
+        const email = (entry.person?.emailAddresses || []).find(email => email.metadata.primary === true)!.value;
+        const name = (entry.person?.names || []).find(name => name.metadata.primary === true)!.displayName;
```

Comment:
do we always expect to have a primary name when we have a primary emailAddress?
Otherwise crash this function?

Reply:
I would not force unwrap `!` any data that comes from external API. As Roman says, it may break things.

Reply:
Right, done 👍